trajopt/include/trajopt/trajectory_costs.hpp

```diff
@@ -48,11 +44,11 @@ class TRAJOPT_API JointPosIneqCost : public sco::Cost
 {
 public:
   /** @brief Forms error in QuadExpr - independent of penalty type */
-  JointPosIneqCost(const VarArray& traj,
-                   const Eigen::VectorXd& coeffs,
-                   const Eigen::VectorXd& targets,
-                   const Eigen::VectorXd& upper_limits,
-                   const Eigen::VectorXd& lower_limits,
+  JointPosIneqCost(VarArray traj,
+                   Eigen::VectorXd coeffs,
+                   Eigen::VectorXd targets,
+                   Eigen::VectorXd upper_limits,
+                   Eigen::VectorXd lower_limits,
```

Comment:
This is very interesting. So basically if it is not a pointer, shared point or fundamental type then you must pass it by reference because you have no idea what libraries use Eigen or not.

Need to go through the library and check for this.